alws/release_planner.py

```diff
@@ -0,0 +1,704 @@
+                    if build_tasks and artifact_task_id not in build_tasks:
+                        continue
+                    artifact_name = rpm.artifact.name
+                    if '.src.' in artifact_name:
+                        src_rpm_names.append(artifact_name)
+                    pkg_info = await self._pulp_client.get_rpm_package(
+                        rpm.artifact.href, include_fields=packages_fields)
+                    pkg_info['artifact_href'] = rpm.artifact.href
+                    pkg_info['full_name'] = artifact_name
+                    build_task = next(
+                        task for task in build.tasks
+                        if task.id == artifact_task_id
+                    )
+                    pkg_info['task_arch'] = build_task.arch
+                    pkg_info['force'] = False
+                    pulp_packages.append(pkg_info)
+            for task in build.tasks:
+                if task.rpm_module and task.id in build_tasks:
+                    key = (
+                        task.rpm_module.name,
+                        task.rpm_module.stream,
+                        task.rpm_module.version,
+                        task.rpm_module.arch
+                    )
+                    if key in modules_to_release:
+                        continue
+                    module_repo = next(
+                        build_repo for build_repo in task.build.repos
+                        if build_repo.arch == task.arch
+                        and not build_repo.debug
+                        and build_repo.type == 'rpm'
+                    )
+                    template = await self._pulp_client.get_repo_modules_yaml(
+                        module_repo.url)
+                    module_index = IndexWrapper.from_template(template)
+                    for module in module_index.iter_modules():
+                        modules_to_release[key] = {
+                            'build_id': build.id,
+                            'name': module.name,
+                            'stream': module.stream,
+                            'version': module.version,
+                            'context': module.context,
+                            'arch': module.arch,
+                            'template': module.render()
+                        }
+        return pulp_packages, src_rpm_names, list(modules_to_release.values())
+
+    async def check_package_presence_in_repo(
+        self,
+        pkgs_nevra: dict,
+        repo_ver_href: str,
+        repo_id: int,
+        arch: str,
+    ):
+        params = {
+            'name__in': ','.join(pkgs_nevra['name']),
+            'epoch__in': ','.join(pkgs_nevra['epoch']),
+            'version__in': ','.join(pkgs_nevra['version']),
+            'release__in': ','.join(pkgs_nevra['release']),
+            'arch': arch,
+            'repository_version': repo_ver_href,
+            'fields': 'pulp_href,name,epoch,version,release,arch',
```

Comment:
fields could be passed as list

Reply:
recently i've found some interesting thing, if we pass in pulp list of str for some params, something going wrong in pulp and query looks like this:
`rpm/packages/?fields=pulp_href&?fields=name&?fields=epoch&?fields=version...`
it's look like a bug or something, and i'm afraid that with large params of lists, in very bad case we can exceed limit of query path